app/src/ui/dialog/dialog.tsx

```diff
@@ -709,6 +710,58 @@ export class Dialog extends React.Component<DialogProps, IDialogState> {
     )
   }
 
+  /**
+   * Gets the aria-labelledby and aria-describedby attributes for the dialog
+   * element.
+   *
+   * The correct semantics are that the dialog element should have the
+   * aria-labelledby and the aria-describedby is optional unless the dialog has
+   * a role of alertdialog, in which case both are required.
+   *
+   * However, macOs Ventura introduced a regression in that:
+   *
+   * For role of 'dialog' (default),  the aria-labelledby is not announced and
+   *    if provided prevents the aria-describedby from being announced. Thus,
+   *    this method will add the aria-labelledby to the aria-describedby in this
+   *    case.
+   *
+   * For role of 'alertdialog', the aria-labelledby is announced but not the
+   *    aria-describedby. Thus, this method will add both to the
+   *    aria-labelledby.
+   *
+   * Neither of the above is semantically correct tho, hopefully, macOs will be
+   * fixed in a future release. The issue is known for macOS versions 13.0 to
+   * the current version of 13.5 as of 2023-07-31.
+   *
+   * A known macOS behavior is that if two ids are provided to the
+   * aria-describedby only the first one is announced with a note about the
+   * second one existing. This currently does not impact us as we only provide
+   * one id for non-alert dialogs and the alert dialogs are handled with the
+   * `aria-labelledby` where both ids are announced.
+   *
+   */
+  private getAriaAttributes() {
```

Comment:
Another option as seen in https://github.com/desktop/desktop/pull/17134 would be to set the `aria-label` attribute to the contents of the `props.title`. This would be better semantically, but, sometimes that title is not a string and therefore using `aria-describedby` allows us to still point to the id of the title for both strings and JSX.elements.  Additionally, the below has less hairy code.